flink-runtime/src/main/java/org/apache/flink/runtime/state/changelog/inmemory/InMemoryStateChangelogWriter.java

```diff
@@ -61,20 +64,26 @@ public SequenceNumber lastAppendedSequenceNumber() {
     }
```

Comment:
> the sqn of appended value starts from 1L

This is controlled by the writer, and in `InMemoryStateChangelogWriter` it's `0L`.
The backend then will request to persist changes from `0L` according to `lastUploadedFrom`  = `materializedTo`.
Does this resolve your concern?

Reply:
It makes sense.
However, it makes me a little confused that sequence number for records in ```InMemoryStateChangelogWriter ``` actually starts from ```1L```, not the ```SequenceNumber.FIRST```. The word FIRST here means an initialized value instead of a valid beginning, right? If so, it does not bother me.

Reply:
That's right and I agree it is confusing.
I'm going to replace `SequenceNumber.FIRST` with a new method `writer.initialSequenceNumber()`. So that writer could use whatever initial SequenceNumber (`0L` in `InMemoryStateChangelogWriter`).